drivers/gpio/gpio_nrfx.c

```diff
@@ -11,6 +11,12 @@
 
 #include "gpio_utils.h"
 
+#if IS_ENABLED(CONFIG_GPIO_NRF_INT_EDGE_USING_SENSE) &&	\
+	!defined(NRF_GPIO_LATCH_PRESENT)
+#error "GPIO LATCH is required by edge interrupts using GPIO SENSE," \
+	"but it is not supported by the platform."
+#endif
+
```

Comment:
We tend to handle this kind of checks with `BUILD_ASSERT` macro. There are multiple examples in `drivers/`.

Reply:
Such construct does not work:
```
BUILD_ASSERT(!IS_ENABLED(CONFIG_GPIO_NRF_INT_EDGE_USING_SENSE) ||
	     defined(NRF_GPIO_LATCH_PRESENT),
	     "GPIO LATCH is required by edge interrupts using GPIO SENSE," \
	     "but it is not supported by the platform.");
```
because we get:
```
/project/zephyr/drivers/gpio/gpio_nrfx.c:15:7: warning: implicit declaration of function 'defined' [-Wimplicit-function-declaration]
   15 |       defined(NRF_GPIO_LATCH_PRESENT),
      |       ^~~~~~~
/project/zephyr/include/toolchain/gcc.h:62:51: note: in definition of macro 'BUILD_ASSERT'
   62 | #define BUILD_ASSERT(EXPR, MSG...) _Static_assert(EXPR, "" MSG)
      |                                                   ^~~~
/project/zephyr/drivers/gpio/gpio_nrfx.c:14:64: error: expression in static assertion is not constant
   14 | BUILD_ASSERT(!IS_ENABLED(CONFIG_GPIO_NRF_INT_EDGE_USING_SENSE) ||
      |              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
   15 |       defined(NRF_GPIO_LATCH_PRESENT),
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                           
/project/zephyr/include/toolchain/gcc.h:62:51: note: in definition of macro 'BUILD_ASSERT'
   62 | #define BUILD_ASSERT(EXPR, MSG...) _Static_assert(EXPR, "" MSG)
      |                                                   ^~~~
```

Should I leave it as is or use `BUILD_ASSERT(0, ...)` instead of `#error`?

Reply:
Let's leave it as is.